### PR TITLE
Improve pizza demo with logic

### PR DESCRIPTION
**Pull request by LawlAoux**

### CODE REVIEW

File: agenda/main.py [hunk 1]
```diff
@@ -28,17 +28,28 @@ async def responder_with_state(request):
...
                 computation_result = await bot({composers.event: request, **state})
                 state = computation_result
-                return state[graph.make_computation_node(composers.utter)]
+                return (
+                    state[graph.make_computation_node(composers.utter)],
+                    gamla.pipe(
+                        state[graph.make_computation_node(composers.debug_states)],
+                        gamla.map(
+                            gamla.when(
+                                gamla.equals(composers.UNKNOWN), gamla.just("Unknown")
```
Comment (uriva) [inline agenda/main.py:44]: maybe `just(None)` and replace it to text in the client side

File: config_to_bot/resolvers.py [hunk 2]
```diff
@@ -117,29 +117,33 @@ def _sentences_similarity(user_utterance: str, examples: Tuple[str, ...]) -> int
 )
 
 
-def _listen_to_bool_or_intent(
+def _listen_to_intent(
     examples: Tuple[str, ...]
 ) -> Callable[[str], Union[bool, agenda.Unknown]]:
     def parse_bool(user_utterance: str):
         if examples and _sentences_similarity(user_utterance, examples) >= 0.9:
```
Comment (uriva) [inline config_to_bot/resolvers.py:124]: `return bool(examples) and _sentences_similarity(user_utterance, examples) >= 0.9`

**Review by uriva — APPROVED** [review]

File: agenda/composers.py [hunk 3]
```diff
@@ -209,6 +215,17 @@ def remember_or_forget(value, memory, should_forget):
     return remember_or_forget
 
 
+@_remove_sinks_and_sources_and_resolve_ambiguity([state])
+def ever(graph):
+    @mark_state
```
Comment (uriva) [inline agenda/composers.py:220]: might be more readable as a composition using a lambda

File: agenda/main.py (hunk 1 above)
Comment (LawlAoux) [reply agenda/main.py:44]: That's what I did at the beginning, but why replacing there if we can just send Unknown which is a keyword that is is understandable from the client?

**Pull request merged**